platforms/android/src/jp/wizcorp/phonegap/plugin/WizAssets/WizAssetsPlugin.java

```diff
@@ -417,131 +416,35 @@ public AsyncDownload(String url, String uri, String filePath, CallbackContext ca
 
         @Override
         protected Void doInBackground(File... params) {
+            File file = null;
             try {
-                File file = new File(this.filePath);
-
-                // Run async download task
-                File dir = file.getParentFile();
-                if (dir == null || !(dir.mkdirs() || dir.isDirectory())) {
-                    Log.e("WizAssetsPlugin", "Error : subdirectory could not be created");
-                    this.callbackContext.error(createDownloadFileError(DIRECTORY_CREATION_ERROR));
-                    return null;
-                }
-
-                Log.d(TAG, "[Downloading] " + file.getAbsolutePath());
-
-                HttpGet httpRequest;
                 try {
-                    URL url = new URL(this.url);
-                    httpRequest = new HttpGet(url.toURI());
-
-                    // Credential check
-                    String credentials = url.getUserInfo();
-                    if (credentials != null) {
-                        // Add Basic Authentication header
-                        httpRequest.setHeader("Authorization", "Basic " + Base64.encodeToString(credentials.getBytes(), Base64.NO_WRAP));
+                    if (!initialized) {
+                        Log.e(TAG, "Plugin not initialized, call initialize");
+                        throw new Exception("Plugin not initialized, call initialize");
                     }
-                } catch (MalformedURLException e) {
-                    this.callbackContext.error(createDownloadFileError(INVALID_URL_ERROR));
-                    return null;
-                } catch (URISyntaxException e) {
-                    this.callbackContext.error(createDownloadFileError(INVALID_URL_ERROR));
-                    return null;
-                }
-
-                HttpResponse response;
-                HttpClient httpclient = new DefaultHttpClient();
-                try {
-                    response = httpclient.execute(httpRequest);
-                    if (response == null) {
+                    file = new File(this.filePath);
+                    URL url = new URL(this.url);
+                    boolean success = HttpToFile.downloadFile(url, file);
+                    if (!success) {
                         this.callbackContext.error(createDownloadFileError(CONNECTIVITY_ERROR));
-                        return null;
                     }
-                    StatusLine statusLine = response.getStatusLine();
-                    if (statusLine == null) {
-                        this.callbackContext.error(createDownloadFileError(HTTP_REQUEST_ERROR, "No status available"));
-                        return null;
-                    }
-                    int statusCode = statusLine.getStatusCode();
-                    if (statusCode < 200 || statusCode > 299) {
-                        this.callbackContext.error(createDownloadFileError(HTTP_REQUEST_ERROR, statusCode));
```

Comment:
Handling HTTP error with status code is a feature we can't afford to lose. Could this be restored somehow?